src/components/engagement/engagement.repository.ts

```diff
@@ -123,7 +134,13 @@ export class EngagementRepository extends CommonRepository {
       ])
       .return<{ dto: UnsecuredDto<LanguageEngagement & InternshipEngagement> }>(
         merge('props', 'changedProps', {
-          __typename: `[l in labels(node) where l in ['LanguageEngagement', 'InternshipEngagement']][0]`,
+          __typename: `
+            replace(
+              [l in labels(node)
+                where l in ['LanguageEngagement', 'InternshipEngagement', 'Deleted_LanguageEngagement', 'Deleted_InternshipEngagement']][0],
+              'Deleted_',
+              ''
+            )`,
```

Comment:
```suggestion
          __typename: stripIndent`
            replace(
              [l in labels(node)
                where l in ['LanguageEngagement', 'InternshipEngagement', 'Deleted_LanguageEngagement', 'Deleted_InternshipEngagement']][0],
              'Deleted_',
              ''
            )
          `,
```

If we keep doing this too we should move it to a function.